ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessorTest.kt

```diff
@@ -177,8 +177,28 @@ class VariableProcessorTest {
     }
 
     @Test
-    fun `process variables processes sub_offer_price`() {
-        expectVariablesResult("{{ sub_offer_price }}", "INTRO_OFFER_PRICE")
+    fun `process variables processes sub_offer_price for free trials`() {
+        expectVariablesResult("{{ sub_offer_price }}", "Free")
+    }
+
+    @Test
+    fun `process variables processes sub_offer_price for intro prices`() {
+        expectVariablesResult("{{ sub_offer_price }}", "$3.99", rcPackage = TestData.Packages.bimonthly)
+    }
+
+    @Test
+    fun `process variables processes sub_offer_price for both free trial and intro price`() {
+        expectVariablesResult("{{ sub_offer_price }}", "Free", rcPackage = TestData.Packages.quarterly)
+    }
+
+    @Test
+    fun `process variables does not process sub_offer_price for inapp products`() {
+        expectVariablesResult("{{ sub_offer_price }}", "{{ sub_offer_price }}", rcPackage = TestData.Packages.lifetime)
```

Comment:
Yeah... In iOS we would just have an empty string correct? I can do that as well... Might still look weird, but maybe better than this.

Reply:
Yeah exactly. As far as the SDK is concerned that's the best we  can do. 
Remember the dashboard validates that these aren't used when they shouldn't.

Reply:
I see you fixed that by changing `String?` to `String`. I think we should avoid that problem for all variables though. You could keep the `String?` result, but make the variable processing convert `nil` to empty strings instead of leaving the variable names in there.

Reply:
I wonder... It feels like the default for each variable type should depend on each method/provider? I mean, we can have a final "catch all nulls and default to empty string" kinda thing, but we already have fallbacks for some of the variables like `sub_period` or `sub_duration` (In iOS we have some fallbacks to empty string at the localization class). But well, I can see moving it there as well, so will do that now 👍

Reply:
> I wonder... It feels like the default for each variable type should depend on each method/provider?

Yeah, in that case each provider can specify a fallback.
But the outer method shouldn't accidentally leak variable names. It should be _impossible_ for a user to accidentally see a variable in the UI.

Reply:
Applied a fallback to empty strings here: https://github.com/RevenueCat/purchases-android/pull/1279/commits/1a101724964d5c0a7f609afe04cad5fa2acc5a06. That way we should catch it in case we can't find a proper value for a variable and log a warning.